src/main/java/com/aws/greengrass/lifecyclemanager/GreengrassService.java

```diff
@@ -495,6 +495,8 @@ public void addOrUpdateDependency(GreengrassService dependencyService, Dependenc
                 context.get(Kernel.class).clearODcache();
                 return new DependencyInfo(dependencyType, isDefault, listener);
             });
+            // Clear cache after updating dependency list
+            context.get(Kernel.class).clearODcache();
```

Comment:
Why do we need to clear again? I thought line 495 already clear it

Reply:
I saw a test fail once which could have been due to a race, so clearing again here to try and avoid that, won't hurt performance since there's such a small likelihood that anyone is querying the list